middleware_timeout_test.go

```diff
@@ -0,0 +1,46 @@
+package gin
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	contractshttp "github.com/goravel/framework/contracts/http"
+	mocksconfig "github.com/goravel/framework/mocks/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTimeoutMiddleware(t *testing.T) {
+	mockConfig := mocksconfig.NewConfig(t)
+	mockConfig.EXPECT().GetBool("app.debug").Return(true).Once()
+	mockConfig.EXPECT().GetInt("http.drivers.gin.body_limit", 4096).Return(4096).Once()
+
+	route, err := NewRoute(mockConfig, nil)
+	require.NoError(t, err)
+
+	route.Middleware(Timeout(1*time.Second)).Get("/timeout", func(ctx contractshttp.Context) contractshttp.Response {
+		time.Sleep(1 * time.Second)
```

Comment:
The test case failed in Windows: https://github.com/goravel/gin/actions/runs/11534959849/job/32113680328?pr=97

We may try to change this to 2 second:

```suggestion
		time.Sleep(2 * time.Second)
```

<img width="770" alt="image" src="https://github.com/user-attachments/assets/25debba4-5ed4-4aa4-9dfc-f2165746fc2e">